pypika/terms.py

```diff
@@ -113,9 +154,39 @@ def wrap_constant(val, wrapper_cls: Optional[Type["Term"]] = None) -> WrappedCon
         wrapper_cls = wrapper_cls or ValueWrapper
         return wrapper_cls(val)
 
+    @overload
+    @staticmethod
+    def wrap_json(val: TermT, wrapper_cls: Optional[Type["ValueWrapper"]] = None) -> TermT:
+        ...
+
+    @overload
+    @staticmethod
+    def wrap_json(val: IntervalT, wrapper_cls: Optional[Type["ValueWrapper"]] = None) -> IntervalT:
+        ...
+
+    @overload
+    @staticmethod
+    def wrap_json(val: None, wrapper_cls: Optional[Type["ValueWrapper"]] = None) -> "NullValue":
+        ...
+
+    @overload
+    @staticmethod
+    def wrap_json(val: Union[str, int, bool], wrapper_cls: Type["ValueWrapperT"]) -> "ValueWrapperT":
+        ...
+
+    @overload
+    @staticmethod
+    def wrap_json(val: Union[str, int, bool], wrapper_cls: None = None) -> "ValueWrapper":
+        ...
+
+    @overload
+    @staticmethod
+    def wrap_json(val: object, wrapper_cls: Optional[Type["ValueWrapper"]] = None) -> "JSON":
+        ...
+
     @staticmethod
     def wrap_json(
```

Comment:
The wrap_json()'s overloads are reported by mypy:
- Overloaded function signatures 1 and 6 overlap with incompatible return types  [misc]
- Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader  [misc] (Rubicon: This signature might be removed since the TermT is including IntervalT, `Interval` is already a `Term`)
- Overloaded function signatures 2 and 6 overlap with incompatible return types  [misc]
- Overloaded function signatures 3 and 6 overlap with incompatible return types  [misc]
- Overloaded function signatures 4 and 6 overlap with incompatible return types  [misc]
- Overloaded function signatures 5 and 6 overlap with incompatible return types  [misc]

Reply:
The rest errors are looked like they could not be resolved with a workaround, may you leave a comment there as a tip?

Reply:
I used mypy to check my code, but I found more issues such as generics, and the "default" parameter in TypeVar. I need to look into these issues over the next few days. Before that, I used pyright (basic mode) to check my code. I'm sorry about that.